### PR TITLE
[master] Add signing information to asset manifest

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Port #41889 to master

This is blocking using post build signing in master.

### CODE REVIEW

**Comment by jkoritzinsky** [comment]
@mmitche how do I start a post build signing enabled test run for an official build?

File: src/installer/publish/prepare-artifacts.proj [hunk 1]
```diff
@@ -1,5 +1,15 @@
...
+  
+  <!-- When doing post build sign, pull in the arcade sign.props file, which will
+       create some initial sign groups and then pulls in the repo's custom Signing.props overrides.
+       Before importing the signing props. Set PrepareArtifactst=true. Depending on context (SDK project vs. not),
+       the initial imports of livebuilds.targets and Configuration.props are already imported and thus cannot
+       be imported again. Without those imports, the globbing of files to sign will not work properly. -->
+  <PropertyGroup>
+    <PrepareArtifacts>true</PrepareArtifacts>
+  </PropertyGroup>
+  <Import Project="../tools/Sign.props" Sdk="Microsoft.DotNet.Arcade.Sdk" />
```
Comment (ViktorHofer) [inline src/installer/publish/prepare-artifacts.proj:12]: oh wow

**Review by ViktorHofer — COMMENTED** [review]
As I don't have enough context on this part of our infra I can't really approve...

**Comment by mmitche** [comment]
> @mmitche how do I start a post build signing enabled test run for an official build?

https://dev.azure.com/dnceng/internal/_build/results?buildId=878115&view=results will do this, but it needs a switch to be ported from the main staging pipeline: https://dev.azure.com/dnceng/internal/_build?definitionId=799

The main staging pipeline is for full SDK builds, while validate is run automatically on a nightly basis.

**Comment by hoyosjs** [comment]
@jkoritzinsky isn't there another PR open for this? I can take a look, but at least we need to run a build without postbuild to ensure no regressions.

**Comment by mmitche** [comment]
@jkoritzinsky There's a PR for this already https://github.com/dotnet/runtime/pull/43896. It does port another PR which is required.

**Comment by jkoritzinsky** [comment]
I couldn't find the other PR when I searched around. I'll close this in favor of that one.

**Pull request closed without merging**